### ISSUE TITLE
State history view is not well suited for multi-dimensional alerts

### ISSUE THREAD

**Issue by JohnnyQQQQ** [grafana/grafana#45900]
Currently, we put all dimensions in a flat list. We should make multiple lists grouped by dimensions so that one can easier scan what's going on.

**Comment by jtheory** [comment]
We'll need to clarify this & then pull in Jess before we can prioritize it

**Comment by joeblubaugh** [comment]
I think this is also closely related to #45300, which involves overlaying state history on a chart. I have an idea about a composite view with state history inside an accordian alongside our chart, instead of our existing state history modal that's difficult to visually connect to the timeseries.

**Comment by joeblubaugh** [comment]
@jessover9000 @gillesdemey and I had a conversation today to talk about some ideas to explore to improve the view.

A few outputs:
Joe and Jess independently came to the idea of using the State History Panel as a starting point for visualizing alert instance state - one instance per row, with the state as a timeline left-to-right.

We discussed whether to continue to put this in a modal or as an inline element. Due to questions of layout, available screen space, and the fact that the page will be changing substantially over the next few versions, we feel that a modal is probably the right approach

We discussed what to do in the "thousands of instances" situation. We're leaning towards setting an upper bound on the number of instances displayed (and probably on the number of instances returned by the API) so we don't slow the frontend to a crawl or make our visualizations useless.

We discussed filtering and aggregation. We probably won't implement this in the first version of the view, but we think it will become important. We'd like to support filtering by label keys and values, and by state. (Under the hood in the database, the state will be a "special label"). When we support aggregation we'd like to support grouping by state, or by a given label key, and outputting a count for those groups for instances in each state. That might be visualized as a timeseries chart of counts instead of a state timeline.

----

I'm going to repurpose this issue as an Epic to track both the UX and backend work that goes into supporting those ideas (sorry @JohnnyQQQQ but I'm going to edit your comments :-P)

**Comment by armandgrillet** [comment]
@joeblubaugh Is this part of the alert state history epic https://github.com/grafana/grafana/issues/48359 or is it its own? We can have improvements and new features in an epic like in https://github.com/grafana/grafana/issues/48943

**Comment by joeblubaugh** [comment]
I don't think we will many many improvements to this view until #48359, just given the availability of people to work on it. I'll link this issue into there.

**Comment by ethirolle** [comment]
@joeblubaugh thank you. I see that https://github.com/grafana/grafana/issues/48359 has a Milestone of 9.1.0, how sure are we that these changes will make it into 9.1.0? Thanks!

**Comment by gillesdemey** [comment]
@ethirolle I have some work in progress that _should_ be able to make it in before the `9.1` release that will make it easier to grok alert state history changes for multi-dimensional alert rules.

Though this will mostly be a stop-gap solution until we can pick up progress on the new history API again.

**Comment by ethirolle** [comment]
@gillesdemey, how sure are you that this will make it into 9.1? Asking for the customer 😄  Thanks!!

**Comment by gillesdemey** [comment]
The PR is reviewed and merged so it's gonna land in 9.1 ✌️
